xgboost_ray/main.py

```diff
@@ -419,100 +570,167 @@ def _train(params: Dict,
     else:
         params["nthread"] = cpus_per_actor
 
-    # Create queue for communication from worker to caller.
-    # Always create queue.
-    queue = Queue()
-
-    # Create remote actors
-    actors = [
-        _create_actor(
+    # This is a callback that handles actor failures.
+    # We identify the rank of the failed actor, add this to a set of
+    # failed actors (which we might want to restart later), and set its
+    # entry in the actor list to None.
+    def handle_actor_failure(actor_id):
+        rank = _actors.index(actor_id)
+        _failed_actor_ranks.add(rank)
+        _actors[rank] = None
+
+    # Here we create new actors. In the first invocation of _train(), this
+    # will be all actors. In future invocations, this may be less than
+    # the num_actors setting, depending on the failure mode.
+    newly_created = 0
+    for i in list(_failed_actor_ranks):
+        if _actors[i] is not None:
+            raise RuntimeError(
+                f"Trying to create actor with rank {i}, but it already "
+                f"exists.")
+        actor = _create_actor(
```

Comment:
Question - from what I understand, failed actors will have their rank added to `_failed_actor_ranks`. When `_train` is called, don't these actors get recreated?

I'm just specifically looking for the situation where a node dies and new actor creation  step is skipped

Reply:
That's in line 848:
```
                 start_actor_ranks = set() 
```
here the set is cleared (I guess we could just `.clear()` it instead). With it being empty, no actors are restarted

Reply:
Ah, got it.